vendor/github.com/grafana/dskit/cache/memcached.go

```diff
@@ -29,21 +22,21 @@ type MemcachedCache struct {
 }
 
 // NewMemcachedCache makes a new MemcachedCache.
-func NewMemcachedCache(name string, logger log.Logger, memcached cacheutil.RemoteCacheClient, reg prometheus.Registerer) *MemcachedCache {
+func NewMemcachedCache(name string, logger log.Logger, memcached RemoteCacheClient, reg prometheus.Registerer) *MemcachedCache {
 	c := &MemcachedCache{
 		logger:    logger,
 		memcached: memcached,
 		name:      name,
 	}
 
 	c.requests = promauto.With(reg).NewCounter(prometheus.CounterOpts{
-		Name:        "thanos_cache_memcached_requests_total",
```

Comment:
This is a breaking change for Mimir. It's fine doing this change in dskit, but then in Mimir you should wrap the registerer to add the `thanos_` prefix back.